archiver.go

```diff
@@ -221,8 +221,11 @@ func openAndCopyFile(file File, w io.Writer) error {
 		return err
 	}
 	defer fileReader.Close()
-	_, err = io.Copy(w, fileReader)
-	return err
+	_, err = io.CopyN(w, fileReader, file.Size())
```

Comment:
Can we add a comment here explaining why we're doing this instead of just io.Copy()? And why we ignore EOF errors below.

Reply:
https://github.com/mholt/archiver/pull/8#issue-169892941

> When file is in use and size getting increased , create compressed file will failed and error is "copying contents: archive/tar: write too long". Changing to use CopyN to copy only the file size at that time, also checks io.EOF if file becomes small.

Reply:
```suggestion
	// When file is in use and size is being written to, creating the compressed
	// file will fail with "archive/tar: write too long." Using CopyN gracefully
	// handles this.
	_, err = io.CopyN(w, fileReader, file.Size())
```